feat: implement hover effects to buttons and links for better interactivity

Marvellous-Udoye opened the pull request:
## Description  
This PR enhances the UI interactivity by adding consistent hover effects to buttons and links across the application. 

## Related Issue  
Related to #65 

## Type of Change  
- [ ] Bug fix / Refactor
- [ ] New feature  
- [ ] Documentation / Content update  
- [x] UI update  

## Checklist  
- [x] I have tested my changes locally  
- [ ] I have updated documentation (if applicable)  
- [x] I have included screenshots or relevant links (if applicable)
- [x] My changes follow the project's code style and contribution guidelines  

⚠️ **Note**: The implemented hover effect applies for other pages and components across the application, not just the ones in the demo.

https://github.com/user-attachments/assets/97fe53bd-5511-47f1-a2a5-c480d394703a

MutiatBash commented:
Hi @Marvellous-Udoye …..I left a comment for you on a file

Marvellous-Udoye commented:
👋 @MutiatBash, i couldn't see any comment on any file.

Marvellous-Udoye commented:
👋 @MutiatBash, i couldn't see any comments on any file.

MutiatBash commented:
Please can I see the hover effect for the links
You should see my comments on files when you scroll up to the beginning of the conversation

MutiatBash commented:
> @MutiatBash
> 
>  2025-06-08.01-36-06.mp4

Hi…..I think for the links it’s better to give them the openest’s blue color and make them maybe bold on hover, then remove the scale effect….while for the Google and GitHub buttons….you can remove the color on hover and just make them scale

Marvellous-Udoye commented:
@MutiatBash 

https://github.com/user-attachments/assets/0b9516b4-4ed5-4aed-b849-11ae9930a44d

Marvellous-Udoye commented:
Affirmative, they redirects properly.
I've added an hover effect to the nav links and updated the hover effect on the links in the about us and resources page.

Marvellous-Udoye commented:
@MutiatBash 

https://github.com/user-attachments/assets/6a7a37dd-ad84-404c-a45f-a5c59d521011

MutiatBash reviewed (approved):
Thank you for your contribution🎉🙌✨
I’ll be merging this
